### PR TITLE
1.17

### PR DESCRIPTION
**Pull request by end-user**
Updates for 1.17, plus some code cleanup

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,86 +1,81 @@
...
 }
-
-sourceCompatibility = JavaVersion.VERSION_1_8
-targetCompatibility = JavaVersion.VERSION_1_8
+repositories {
+    maven { url 'https://maven.fabricmc.net/' }
+    maven { url 'https://maven.shedaniel.me/' }
+    maven { url 'https://maven.terraformersmc.com/releases/'}
+}
+sourceCompatibility = JavaVersion.VERSION_16
```
Comment (flier268) [inline build.gradle:10]: Java 1.6?

File: build.gradle [hunk 2]
```diff
@@ -1,86 +1,81 @@
...
+targetCompatibility = JavaVersion.VERSION_16
```
Comment (flier268) [inline build.gradle:11]: Java 1.6?

**Review by flier268 — COMMENTED** [review]
Thank for @end-user 
I'm a C# developer, so i am not good at Java. This mod is my toy, so I don't have a lot of time to upgrade it.
Thank you for review my code.
Is there any reason to choose Java 1.6 ?

File: build.gradle (hunk 2 above)
Comment (end-user) [reply build.gradle:11]: Not 1.6 - 16.  MC 1.17 [requires Java 1.16](https://minecraft.fandom.com/wiki/Java_Edition_21w19a), and conveniently Gradle 7 does too.

**Pull request merged**